packages/bridge-ui/src/components/buttons/Connect.svelte

```diff
@@ -11,22 +10,74 @@
     ConnectorNotFoundError,
     getNetwork,
     getAccount,
+    Client,
+    createClient,
+    configureChains,
   } from '@wagmi/core';
+  import { CoinbaseWalletConnector } from '@wagmi/core/connectors/coinbaseWallet';
+  import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect';
+  import { MetaMaskConnector } from '@wagmi/core/connectors/metaMask';
+  import { publicProvider } from '@wagmi/core/providers/public';
+  import { jsonRpcProvider } from '@wagmi/core/providers/jsonRpc';
 
-  import { CHAIN_MAINNET, CHAIN_TKO } from '../../domain/chain';
+  import { signer } from '../../store/signer';
+  import {
+    CHAIN_MAINNET,
+    CHAIN_TKO,
+    mainnet,
+    taiko,
+    providers,
+  } from '../../domain/chain';
   import { fromChain, toChain } from '../../store/chain';
   import {
     isSwitchEthereumChainModalOpen,
     isConnectWalletModalOpen,
   } from '../../store/modal';
   import { errorToast, successToast } from '../../utils/toast';
   import Modal from '../modals/Modal.svelte';
-  import { wagmiClient } from '../../store/wagmi';
   import MetaMask from '../icons/MetaMask.svelte';
   import WalletConnect from '../icons/WalletConnect.svelte';
   import CoinbaseWallet from '../icons/CoinbaseWallet.svelte';
   import { transactioner, transactions } from '../../store/transactions';
 
+  const iconMap = {
```

Comment:
Not sure if initialising `wagmi` in a UI component file makes sense. It should either be in `App.svelte` like it was and I am fine with not writing it to a store, or it could be exposed as a singleton from service.

It makes more semantic sense to have components act as consumers and not initialisers. @jscriptcoder @cyberhorsey  what do you think about this?

Reply:
Agree the component should act as a consumer, in which case the client should be injected as a prop. The problem with using Svelte store in order to provide Connect component with wagmi client feels hacky. I prefer the idea of using a service like you mention, but definitely I don't see it as part of the state of the application

Reply:
I think its fine to have here, but maybe the logic for instantiating the wagmi stuff could be wrapped in a utils method, so the definition is not here, but the consuming of it is? I dont think it merits being its own service, but a simple `utils/connectWallet.ts` where it wraps all the wagmi/connect/init logic could be a lot cleaner.